bionty/_source.py

```diff
@@ -9,11 +9,29 @@
 def get_source_record(
     registry: type[SQLRecord],
     organism: str | SQLRecord | None = None,
-    source: SQLRecord | None = None,
+    source: SQLRecord | str | None = None,
+    version: str | None = None,
```

Comment:
Here we should only accepts `source` as a record which contains the version (source.version), so I'd revert it.